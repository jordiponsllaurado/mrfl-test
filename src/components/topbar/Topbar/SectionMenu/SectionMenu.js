import PropTypes from 'prop-types';
import styled from 'styled-components';

const SectionWrapper = styled.a`
  ${({ color }) => color && `color: ${color};`}
  ${({ textSize }) => textSize && `font-size: ${textSize};`}
  ${({ active, underlineColor }) => active && `border-bottom: 3px solid ${underlineColor}`}
`;

const SectionMenu = ({ sections, sectionActive, onChange, color, textSize }) => {
  const renderItem = item => {
    const className = `section-item${sectionActive === item ? ' active' : ''}`;

    const handleChange = e => {
      onChange(e.target.innerHTML);
    };

    return (
      <SectionWrapper
        key={item}
        className={className}
        active={sectionActive === item ? 'active' : ''}
        onClick={handleChange}
        color={color}
        underlineColor={color || 'orange'}
        textSize={textSize}
      >
        {item}
      </SectionWrapper>
    );
  };

  return <div className="section-menu">{sections.map(renderItem)}</div>;
};

SectionMenu.propTypes = {
  sections: PropTypes.arrayOf(PropTypes.string).isRequired,
  sectionActive: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  color: PropTypes.string,
  textSize: PropTypes.string
};

SectionMenu.defaultProps = {
  color: '',
  textSize: '22px'
};
export default SectionMenu;
