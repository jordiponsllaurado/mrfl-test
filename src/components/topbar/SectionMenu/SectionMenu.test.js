import { render } from '@testing-library/react';
import SectionMenu from './SectionMenu';

const sections = ['facilitate', 'administration', 'invoice', 'bandwidth'];
const onChangeSection = jest.fn();

beforeEach(() => {
  onChangeSection.mockClear();
});

test('renders Section Menu', () => {
  const { container } = render(
    <SectionMenu sections={sections} sectionActive="administration" onChange={onChangeSection} />
  );
  expect(container).toMatchSnapshot();
});

test('renders Section Menu with customizations', () => {
  const { container } = render(
    <SectionMenu
      sections={sections}
      sectionActive="facilitate"
      onChange={onChangeSection}
      color="green"
      textSize="12px"
    />
  );
  expect(container).toMatchSnapshot();
});
