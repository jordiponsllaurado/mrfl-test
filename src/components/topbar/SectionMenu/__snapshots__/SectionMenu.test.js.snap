// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Section Menu 1`] = `
<div>
  <div
    class="section-menu"
  >
    <a
      class="sc-bdfBwQ dBSomw section-item"
      color=""
    >
      facilitate
    </a>
    <a
      class="sc-bdfBwQ eHIKPq section-item active"
      color=""
    >
      administration
    </a>
    <a
      class="sc-bdfBwQ dBSomw section-item"
      color=""
    >
      invoice
    </a>
    <a
      class="sc-bdfBwQ dBSomw section-item"
      color=""
    >
      bandwidth
    </a>
  </div>
</div>
`;

exports[`renders Section Menu with customizations 1`] = `
<div>
  <div
    class="section-menu"
  >
    <a
      class="sc-bdfBwQ EUqGI section-item active"
      color="green"
    >
      facilitate
    </a>
    <a
      class="sc-bdfBwQ bSISot section-item"
      color="green"
    >
      administration
    </a>
    <a
      class="sc-bdfBwQ bSISot section-item"
      color="green"
    >
      invoice
    </a>
    <a
      class="sc-bdfBwQ bSISot section-item"
      color="green"
    >
      bandwidth
    </a>
  </div>
</div>
`;
